### PR TITLE
Update dependencies and fix TS errors

### PR DESCRIPTION
**Pull request by gins3000**
Hi, I have been looking around for a light weight TypeScript lib for AR, and the only one I could find (besides [ArgonJS](https://github.com/argonjs/argon), which was certainly not lightweight enough for my tastes) was yours.

Unfortunately I quickly ran into build errors, and I noticed that there has not been a lot of activity in this repository lately. 
Instead of nagging you with an Issue, I decided to update all dependencies, set TypeScript to 'strict', fix npm audit issues and fix thusly broken TypeScript so that it works for my project and send you a PR.

I also updated the prettier configs, since there were some strange side effects, like it reformatting the package.json. Now it runs automatically after build:prod.

I tried not to modify too much, so if there is anything you deem an unnecessary, risky or even wrong change, I'll gladly fix that.
If you don't have time for this, i'll survive, don't worry ;)

Thanks for this, I hope it'll prove useful in my little pet project, one way or another, and that this PR is of any use to you.

Cheers!

### CODE REVIEW

File: docs/classes/barcodemarker.html [hunk 1]
```diff
@@ -118,7 +118,7 @@ <h3>constructor</h3>
 							<aside class="tsd-sources">
 								<p>Overrides BaseMarker.__constructor</p>
 								<ul>
-									<li>Defined in <a href="https://github.com/JamesLMilner/THREEAR/blob/6620e61/src/BarcodeMarker.ts#L23">BarcodeMarker.ts:23</a></li>
```
Comment (JamesLMilner) [inline docs/classes/barcodemarker.html:121]: I think this should still point at the main repo

File: docs/classes/barcodemarker.html [hunk 2]
```diff
@@ -140,7 +140,7 @@ <h3>barcode<wbr>Value</h3>
 					<div class="tsd-signature tsd-kind-icon">barcode<wbr>Value<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">number</span></div>
 					<aside class="tsd-sources">
 						<ul>
-							<li>Defined in <a href="https://github.com/JamesLMilner/THREEAR/blob/6620e61/src/BarcodeMarker.ts#L22">BarcodeMarker.ts:22</a></li>
```
Comment (JamesLMilner) [inline docs/classes/barcodemarker.html:143]: Same as above, won't comment any more as these are all autogenerated but hope that makes sense!

**Comment by JamesLMilner** [comment]
Hey! Thanks so much for this PR - happy to look at merging it. It all looks sensible to me, only concern is that the docs now point at your GitHub rather than mine, is there a chance you could point it back to JamesLMilner? Thanks!

**Review by JamesLMilner — CHANGES_REQUESTED** [review]

File: docs/classes/barcodemarker.html (hunk 2 above)
Comment (gins3000) [reply docs/classes/barcodemarker.html:143]: Oh right, I didn't check the docs. Good thing you did, and thanks for reviewing in the first place.
I'll change these back quickly.

**Pull request merged**